core/perf_test/PerfTest_ViewCopy.hpp

```diff
@@ -42,220 +42,134 @@
-    }
-    Kokkos::fence();
-    time_raw = timer.seconds() / R;
-  }
-#endif
-  double size = 1.0 * N8 * 8 / 1024 / 1024;
-  printf("   Raw:   %lf s   %lf MB   %lf GB/s\n", time_raw, size,
-         2.0 * size / 1024 / time_raw);
-  printf("   Rank7: %lf s   %lf MB   %lf GB/s\n", time7, size,
-         2.0 * size / 1024 / time7);
+  Kokkos::View<double*****, LayoutA> a("A5", N2, N2, N1, N1, N2);
+  Kokkos::View<double*****, LayoutB> b("B5", N2, N2, N1, N1, N2);
+
+  deepcopy_view(a, b, state);
 }
 
 template <class LayoutA, class LayoutB>
-void run_deepcopyview_tests8(int N, int R) {
-  const int N1 = N;
+static void ViewDeepCopy_Rank6(benchmark::State& state) {
+  const int N1 = state.range(0);
   const int N2 = N1 * N1;
-  const int N4 = N2 * N2;
-  const int N8 = N4 * N4;
-
-  double time8, time_raw = 100000.0;
-  {
-    Kokkos::View<double********, LayoutA> a("A8", N1, N1, N1, N1, N1, N1, N1,
-                                            N1);
-    Kokkos::View<double********, LayoutB> b("B8", N1, N1, N1, N1, N1, N1, N1,
-                                            N1);
-    time8 = deepcopy_view(a, b, R) / R;
-  }
-#if defined(KOKKOS_ENABLE_CUDA_LAMBDA) || !defined(KOKKOS_ENABLE_CUDA)
-  {
-    Kokkos::View<double*, LayoutA> a("A1", N8);
-    Kokkos::View<double*, LayoutB> b("B1", N8);
-    double* const a_ptr       = a.data();
-    const double* const b_ptr = b.data();
+
+  Kokkos::View<double******, LayoutA> a("A6", N2, N1, N1, N1, N1, N2);
+  Kokkos::View<double******, LayoutB> b("B6", N2, N1, N1, N1, N1, N2);
+
+  deepcopy_view(a, b, state);
+}
+
+template <class LayoutA, class LayoutB>
+static void ViewDeepCopy_Rank7(benchmark::State& state) {
+  const int N1 = state.range(0);
+  const int N2 = N1 * N1;
+
+  Kokkos::View<double*******, LayoutA> a("A7", N2, N1, N1, N1, N1, N1, N1);
+  Kokkos::View<double*******, LayoutB> b("B7", N2, N1, N1, N1, N1, N1, N1);
+
+  deepcopy_view(a, b, state);
+}
+
+template <class LayoutA, class LayoutB>
+static void ViewDeepCopy_Rank8(benchmark::State& state) {
+  const int N1 = state.range(0);
+
+  Kokkos::View<double********, LayoutA> a("A8", N1, N1, N1, N1, N1, N1, N1, N1);
+  Kokkos::View<double********, LayoutB> b("B8", N1, N1, N1, N1, N1, N1, N1, N1);
+
+  deepcopy_view(a, b, state);
+}
+
+template <class LayoutA, class LayoutB>
+static void ViewDeepCopy_Raw(benchmark::State& state) {
+  const int N8 = std::pow(state.range(0), 8);
+
+  Kokkos::View<double*, LayoutA> a("A1", N8);
+  Kokkos::View<double*, LayoutB> b("B1", N8);
+  double* const a_ptr       = a.data();
+  const double* const b_ptr = b.data();
+
+  for (auto _ : state) {
     Kokkos::Timer timer;
-    for (int r = 0; r < R; r++) {
-      Kokkos::parallel_for(
-          N8, KOKKOS_LAMBDA(const int& i) { a_ptr[i] = b_ptr[i]; });
-    }
+    Kokkos::parallel_for(
+        N8, KOKKOS_LAMBDA(const int& i) { a_ptr[i] = b_ptr[i]; });
     Kokkos::fence();
-    time_raw = timer.seconds() / R;
+
+    report_results(state, timer.seconds());
   }
```

Comment:
This means that we are measuring something slightly different, right? Previously, we were enqueuing a bunch of kernels and waiting on them and now we are fencing after every kernel. Compared with the changes above, this is at least more of an apple-to-apple comparison now.
It doesn't really make sense to fence after the loop, so the changes here are good.

Reply:
As mentioned above, switched to measuring single kernel run and letting `benchmark` take care of the rest.